traffic_ops/traffic_ops_golang/server/servers_assignment.go

```diff
@@ -61,18 +62,21 @@ func AssignDeliveryServicesToServerHandler(w http.ResponseWriter, r *http.Reques
 		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, err, nil)
 		return
 	}
-
-	assignedDSes, err := assignDeliveryServicesToServer(server, dsList, replace, inf.Tx.Tx)
+	serverName, ok, err := dbhelpers.GetServerNameFromID(inf.Tx.Tx, int64(server))
 	if err != nil {
-		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, err)
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusInternalServerError, nil, errors.New("getting server name from ID: "+err.Error()))
 		return
+	} else if !ok {
+		api.HandleErr(w, r, inf.Tx.Tx, http.StatusBadRequest, nil, errors.New("no server with that ID found"))
```

Comment:
Shouldn't this be using `http.StatusNotFound`? Also, 'server not found' might be useful for the user to read, so that they know it failed because the _server_ wasn't found, not because the _deliveryservice_ wasn't found.